packages/react-stand-in/src/createClassProxy.js

```diff
@@ -63,10 +63,13 @@ function createClassProxy(InitialComponent, proxyKey, wrapResult = identity) {
     render() {
       inject(this, proxyGeneration, injectedMembers)
 
-      const result = isFunctionalComponent
+      let result = isFunctionalComponent
         ? CurrentComponent(this.props, this.context)
         : CurrentComponent.prototype.render.call(this)
-
+      
+      if (!!result && !result.$$typeof && result.render){
+        result = result.render()
```

Comment:
I'm not sure this is right. I don't think we should e.g. be creating a new Relay component instance every time in `render`. And anyway this should use `this` as context as above.

Reply:
We cant place such `secret-knowledge` based logic here.
As long it may fix Relay - it can break something else. Or not fix something else.

Reply:
This isn't right though. \:

Constructing a new instance of the component is unlikely to work in the general case.

Maybe just throw an error here and tell the user to install their own monkeypatch per https://github.com/gaearon/react-hot-loader/issues/775#issuecomment-356508850?

Reply:
We dont have any reasons to throw an error. I'll expect a lot of false positives here.
Just give us a bit of time to find a proper way to solve this case as a broad problem, not bound to Relay case.